kroxylicious-app/src/main/resources/log4j2.yaml

```diff
@@ -46,3 +46,8 @@ Configuration:
         additivity: false
         AppenderRef:
           - ref: STDOUT
+      - name: com.github.benmanes.caffeine.cache.LocalAsyncCache
```

Comment:
This block is to disable WARN level logging with stack traces when there is an exception thrown during async cache loading. The class produces no ERROR logs. Caffeine in general looks to take the approach of [minimal logging](https://github.com/ben-manes/caffeine/issues/456#issuecomment-694378232) except in these async loading cases where a failure could be invisible to the user, due to the user not logging the result of exceptionally completed futures. These failures flow into our logging.

Caffeine does support micrometer [[1](https://javadoc.io/doc/io.micrometer/micrometer-core/latest/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetrics.html)][[2](https://javadoc.io/doc/io.micrometer/micrometer-core/latest/io/micrometer/core/instrument/binder/cache/CaffeineStatsCounter.html)], so we should make a ticket to enable this to be switched on.

Reply:
Does SLF4J have a `FATAL` level, I'm kinda on the fence about considering `ERROR` to be disabled. I guess it makes sense that we would still want to capture `ERROR` logs if they were to start being emitted by caffine.

Just to write down for posterity the discussion we had on a call. We already attach to the future and capture the failures so the logging from caffine is redundant to us.

Reply:
Slf4j has FATAL yes, rarely seen it though `
The FATAL level designates very severe error events that will presumably lead the application to abort.`.

`LocalAsyncCache` only produces WARN logs (currently), there are other usages of platform logging in caffeine but this is the only one we know would log in our failed KMS operation case. Caffeine logging in general would be going to the root logger, just this one logger is targeted.

Reply:
I don't understand the context of when we are seeing this exception.  However this [thread](https://github.com/ben-manes/caffeine/issues/885#issuecomment-1470231702) makes me wonder if we improve our exception handling, can we avoid relying on their logging for regular use-cases?

I am surprised that Caffeine doesn't have an error listener.. That seems odd to me.

Reply:
> relying on their logging for regular use-cases
We aren't, we chain off their futures and when they fail we handle it and log in ResilientKms or eventually in FilterHandler if it causes the whole stage to fail. This change silences the redundant logging from within caffeine.

Reply:
The context of when this would happen is, we instantiate an `AsyncLoadingCache` with a loader function like:

```
this.decryptorCache = Caffeine.newBuilder()
                .buildAsync((edek, executor) -> makeDecryptor(edek));
 ```
 
 where `makeDecryptor` returns `CompletableFuture<AesGcmEncryptor>`. Calling the loader function is caffeines responsibility when a client calls `CompletableFuture<V> get(K key);`. Caffeine takes care of ensuring we have one future in flight for each key. If the future supplied by the loader function fails, all the futures from `get` are failed as well. But if the user doesn't handle the exceptionally completed future properly, and log etc then it could obscure the problem.